node/actors/consensus/src/leader/tests.rs

```diff
@@ -25,17 +23,16 @@ async fn replica_commit() {
             .sign_msg(validator::ConsensusMsg::ReplicaCommit(
                 validator::ReplicaCommit {
                     view: consensus.leader.view,
-                    proposal_block_hash,
-                    proposal_block_number: validator::BlockNumber(42),
+                    proposal: rng.gen(),
                 },
             ));
 
-    assert_eq!(
-        consensus.leader.process_replica_commit(
-            ctx,
-            &consensus.inner,
-            test_replica_msg.cast().unwrap()
-        ),
-        Err(Error::ReplicaCommitMissingProposal)
-    );
+    match consensus.leader.process_replica_commit(
+        ctx,
+        &consensus.inner,
+        test_replica_msg.cast().unwrap(),
+    ) {
+        Err(super::replica_commit::Error::UnexpectedProposal) => {}
+        res => panic!("unexpected result {res:?}"),
+    }
```

Comment:
Nit: Could be expressed via `assert_matches` (the corresponding crate is already used in the workspace).